docs/configuration.md

```diff
@@ -179,6 +180,24 @@ Date and time in the format `yyyy-MM-ddTHH:mm:ss` (eg `commits-before:
 2015-10-23T12:23:15`) to setup an exclusion range. Effectively any commit before
 `commits-before` will be ignored.
 
+### merge-messages
+Custom merge message formats to enable identification of merge messages that do not
+follow the built-in conventions.  Entries should be added as key-value pairs where
+the value is a regular expression.
+e.g.
+
+```
+merge-messages:
+    tfs: ^Merged (?:PR (?<PullRequestNumber>\d+)): Merge (?<SourceBranch>.+) to (?<TargetBranch>.+)
```

Comment:
How is the `tfs` key defined and used and which other keys do you envision that we add?

Reply:
I don't envisage the project adds _any_ more at all.  The intent was that the user would add entries as needed.  This means new or modified external services don't require a new version of Gitversion to support them.

The key is intended to be an identity so that the match used for the commit can be logged or debugged more easily.  I've not implemented this for logging, but can pick that up.

Reply:
@asbjornu I've added a line to log the matching merge message definition in the `MergeMessageBaseVersionStrategy`